src/main/java/TaskList.java

```diff
@@ -0,0 +1,145 @@
+import java.util.ArrayList;
+import java.util.Collections;
+
+/**
+ * Represents the object that stores the user's task list as a static ArrayList of Task objects.
+ * Supports operations to manipulate it's taskList.
+ */
+public class TaskList {
+    private static ArrayList<Task> taskList;
```

Comment:
You might want to remove static from this attribute since you initiate it on the creation of a TaskList. Eventhough your code would still work since we only maintain one class list.